presto-tests/src/main/java/io/prestosql/tests/datatype/CreateAndPrestoInsertDataSetup.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.datatype;
+
+import com.google.common.base.Joiner;
+import io.prestosql.tests.sql.PrestoSqlExecutor;
+import io.prestosql.tests.sql.SqlExecutor;
+import io.prestosql.tests.sql.TestTable;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.IntStream.range;
+
+public class CreateAndPrestoInsertDataSetup
+        implements DataSetup
+{
+    private final SqlExecutor sqlExecutor;
+    private final PrestoSqlExecutor prestoSqlExecutor;
+    private final String tableNamePrefix;
+
+    public CreateAndPrestoInsertDataSetup(SqlExecutor sqlExecutor, PrestoSqlExecutor prestoSqlExecutor, String tableNamePrefix)
+    {
+        this.sqlExecutor = sqlExecutor;
+        this.prestoSqlExecutor = prestoSqlExecutor;
+        this.tableNamePrefix = tableNamePrefix;
+    }
+
+    @Override
+    public TestTable setupTestTable(List<DataTypeTest.Input<?>> inputs)
+    {
+        TestTable testTable = createTestTable(inputs);
```

Comment:
You can use try-with-resouces

```
try (TestTable testTable = createTestTable(inputs)) {
    insertRows(testTable, inputs);
    return testTable;
}
```

Reply:
I don't think I can do this. `testTable` must survive beyond this block.

Reply:
Oh, that's right. I just saw the `close()` in the catch block and missed that.